### ISSUE TITLE
Color-indicate visible tracks and folders, and easier checkbox-control visibility

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#18024]
**(1) Visibility indication**

In addition to the rather complex track filtering planned here https://github.com/osmandapp/OsmAnd-Issues/issues/1889 we are currently lacking a rather trivial indication on the My Places > Tracks screen showing which of the tracks are visible and which of the folders are fully or partially visible on the map. Let's simply dye the corresponding track / folder icon (perhaps also label text) in e.g  green for "Visible" and orange for "Partiallly visible" (only applicable to folders, of course).

This could later be enhanced by a filter button to filter the list to only the colored items, to facilitate a better overview for users with many folders. (Which could replace the current "Visible tracks" screen.)

**(2) Usable track/folder visibility management**

Overall, I am still struggling with this use case, due to various shortfalls:

I have about 70 track folders grouping different scientific/geological classifications. I often have e.g. 10 of these folders fully displayed. Now taking one of them out of the visibility list, i.e. going back to 9, is a real challenge. I wish I somehow could put the entire list structure in a "Visible" selection mode )where checkboxes indicate current visibility status), simply deselect the checkbox in front of one of the visible folders (and/or add some other folders to the visible ones), confirm the new overall  "Visible" selection, and be done.

Right now, I have to go a second "Visible Tracks" screen which also in now way facilitates any of these rather basic tasks actions easily (and unnecessarily has a different arrangement and behavior) ...

**Comment by sonora** [comment]
Strawmen:

<table style="width:100%">
  <tr>
    <th>Regular List View</th>
    <th>Visibility Selection Mode</th>
  </tr>
  <tr>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/845735/dd1ea3c5-f875-4ed0-aa27-16602b488eb7"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/845735/2e0575e6-ed41-4145-a411-c1d2fd3bcab9"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by Rowin63** [comment]
And I want add one wish here: 
what I´m missing is a "select all / deselect all" by folder in the "configure map / tracks" section.
(selecting folders is only possible in "my places/tracks", but, as you pointed out, there I don´t see, which ones are already selected/displayed)

**Comment by sonora** [comment]
@Rowin63 Yes exactly, that would be part of (2): You can toggle the checkbox of each folder to select all and deselect all tracks contained in the folder.

(If you want to select only some of a folder's tracks, you first tap on the folder name (which changes to that folder's content), then tap the checkboxes of individual tracks or sub-folders contained).

**Comment by vshcherb** [comment]
Can't say yes or no, too many features floating around selection, so it needs to be thought through. I think diferent colors for checkbox is no go, way to complicated. I think original request was easier to understand https://github.com/osmandapp/OsmAnd/issues/17103.

Though this request won't be implemented without proper understanding and checkx for folder selection, we need to put reasonable limits in order to avoid too many crashes with OOM but first we lower the limit for OOM.
